pkg/indexer/reorgHandler.go

```diff
@@ -0,0 +1,152 @@
+package indexer
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/xmtp/xmtpd/pkg/blockchain"
+	"github.com/xmtp/xmtpd/pkg/db/queries"
+)
+
+type ChainReorgHandler interface {
+	FindReorgPoint(detectedAt uint64) (uint64, []byte, error)
+}
+
+type ReorgHandler struct {
+	ctx     context.Context
+	client  blockchain.ChainClient
+	queries *queries.Queries
+}
+
+var (
+	ErrNoBlocksFound = errors.New("no blocks found")
+	ErrGetBlock      = errors.New("failed to get block")
+)
+
+// TODO: Make this configurable?
```

Comment:
Rich always says that TODO should have your name in them :) So we know who thought that there is some work left. You will see `TODO(rich)` and `TODO(mkysel)` through the codebase. I don't have strong opinions on this.